extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -314,6 +316,33 @@ export class AnalyticsRoot {
     });
   }
 
+  /**
+   * Check if the given element has a data-vars attribute.
+   * Warn if not the case.
+   * @param {!Element} element
+   * @param {!Array<!Element>} elementArray
+   * @param {string} selector
+   */
+  checkElementDataVars(element, elementArray, selector) {
+    const dataVarKeys = Object.keys(
+      getDataParamsFromAttributes(
+        element,
+        /* computeParamNameFunc */ undefined,
+        VARIABLE_DATA_ATTRIBUTE_KEY
+      )
+    );
+    if (dataVarKeys.length) {
+      elementArray.push(element);
+    } else {
+      user().warn(
```

Comment:
When multiple elements are selected. I think the message here will get duplicated. 

One option is to also print the element information. A better option is to only print the warning message once with the number of elements that don't have `data-vars-` attribute specified. WDYT?

Reply:
Yes I agree, changed up the method to do slightly more work.